### ISSUE TITLE
Send amount with floating point number displays incorrectly in confirmation component for several languages (see description) [flaky]

### ISSUE THREAD

**Issue by Andrey-Roshchupkin** [safe-global/safe-wallet-web#4026]
## Bug description
When attempting to send tokens using the "New Transaction" feature, the amount displayed in the confirmation form is incorrect. Instead of displaying "ETH 0.02," it shows "ETH не число" (and other incorrect variants) for languages that use the comma sign instead of the dot in floating point numbers. This issue occurs specifically in the confirmation component, while other components display the amount correctly.
**Note.** This bug could be flaky because, after some page reloads, the send amount displays correctly.
## Environment
 - Browser: Chrome
 - Website language: Russian (navigator.language = "ru")
 - Wallet: MetaMask
 - Chain: Sepolia testnet

## Steps to reproduce
Precondition: A test Safe with 0.04 SepoliaETH and a MetaMask Wallet with two connected accounts, each holding 0.1 SepoliaETH. Register the Safe wallet with the first account; add the second account. Two confirmations should be set for transaction approval.
1. Open https://app.safe.global
2. Click "New Transaction"
3. Choose "Send tokens"
4. Paste the first MetaMask account address into the "Recipient address or ENS" field
5. Input "0.02" value into the "amount" field and click "Next"
6. Click "Sign" in the confirmation form
7. Sign in the MetaMask widget and change to the second account in MetaMask Wallet
8. Click "Transactions"
9. Click "Confirm" in the transaction row

## Expected result
The confirmation form should display the send amount as "ETH 0.02".
## Obtained result
The confirmation form displays the send amount as "ETH не число".
## Screenshots
![image](https://github.com/user-attachments/assets/241766d8-ac6f-4e04-a738-4e3a433f6dcb)
![image](https://github.com/user-attachments/assets/dc0ca339-d2c4-4d70-8df6-04eb682b39d0)

The same bug with the Finnish language:

![image](https://github.com/user-attachments/assets/4d961996-636e-4a64-a4d1-beb9f425f2bd)
![image](https://github.com/user-attachments/assets/5e034dd6-2d54-4437-a72c-3617b1a02846)
## Probably reasons
The app uses the Intl.NumberFormat constructor to create locale-specific number formatting. The representation of decimal separators (dot or comma) depends on the specified locale.

<details><summary>Locales Using a Dot (.)</summary>
<ul>
<li>en-US (United States)</li>
<li>en-GB (United Kingdom)</li>
<li>en-CA (Canada)</li>
<li>en-AU (Australia)</li>
<li>en-IN (India)</li>
<li>ja-JP (Japan)</li>
<li>zh-CN (China)</li>
<li>ko-KR (South Korea)</li>
<li>en-SG (Singapore)</li>
<li>en-HK (Hong Kong)</li>
<li>en-NZ (New Zealand)</li>
<li>en-ZA (South Africa)</li>
<li>en-PH (Philippines)</li>
<li>en-MY (Malaysia)</li>
</ul>
</details>
<details><summary>Locales Using a Comma (,)</summary>
<ul>
<li>de-DE (Germany)</li>
<li>fr-FR (France)</li>
<li>es-ES (Spain)</li>
<li>it-IT (Italy)</li>
<li>nl-NL (Netherlands)</li>
<li>pt-PT (Portugal)</li>
<li>fi-FI (Finland)</li>
<li>nb-NO (Norway)</li>
<li>sv-SE (Sweden)</li>
<li>da-DK (Denmark)</li>
<li>ru-RU (Russia)</li>
<li>pl-PL (Poland)</li>
<li>cs-CZ (Czech Republic)</li>
<li>hu-HU (Hungary)</li>
<li>ro-RO (Romania)</li>
<li>tr-TR (Turkey)</li>
<li>el-GR (Greece)</li>
<li>pt-BR (Brazil)</li>
<li>es-AR (Argentina)</li>
<li>es-CL (Chile)</li>
<li>es-CO (Colombia)</li>
<li>fr-CA (French-speaking Canada, like Quebec)</li>
<li>nl-BE (Belgium)</li>
<li>fr-BE (Belgium)</li>
</ul>
</details>

It seems that after reloading the page, the bug doesn't appear because the confirmation component uses a memoized value that displays correctly. However, before that, the value formatted previously and correctly displayed in other components is reformatted. If the value was initially formatted with a comma, reformatting it may result NaN in the specified language.

Below is the code that explains how this bug could exist:

```js
const localeRu = "ru";
const localeEn = "en";
const localeFi = "fi";

const value = 20000000000000000;

const amount = value / 1000000000000000000; // 0.02

const getNumberFormatter = (locale, maximumFractionDigits, compact) => {
  return new Intl.NumberFormat(locale, {
    style: "decimal",
    maximumFractionDigits,
    notation: compact ? "compact" : "standard",
  });
};

const numberEn = getNumberFormatter(localeEn, 3).format(amount); // 0.02
const numberRu = getNumberFormatter(localeRu, 3).format(amount); // 0,02
const numberFi = getNumberFormatter(localeFi, 3).format(amount); // 0,02

console.log(numberEn, " -> ", getNumberFormatter(localeEn, 3).format(numberEn)); // 0.02  ->  0.02
console.log(numberRu, " -> ", getNumberFormatter(localeRu, 3).format(numberRu)); // 0,02  ->  не число
console.log(numberFi, " -> ", getNumberFormatter(localeFi, 3).format(numberFi)); // 0,02  ->  epäluku```

**Comment by katspaugh** [comment]
Thanks for the detailed report!

**Comment by katspaugh** [comment]
Tested and couldn't reproduce it anymore. Looks like it has been already fixed in #3883.

Cheers!
